packages/query-core/src/query.ts

```diff
@@ -560,6 +569,11 @@ export class Query<
             ...state,
             isInvalidated: true,
           }
+        case 'meta':
+          return {
+            ...state,
+            meta: action.meta,
+          }
```

Comment:
I think this is too generic. How about an action that says `'pageFetched'` or something that increases the current state? and we'd re-set the counter to zero when a `'fetch'` action is initiated

Reply:
That actually sound very good I will do that